bundle/src/test/java/com/adobe/acs/commons/wcm/comparisons/impl/PageCompareDataImplTest.java

```diff
@@ -0,0 +1,208 @@
+import java.util.Set;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PageCompareDataImplTest {
+
+    @Test
+    public void shouldInitialize() throws Exception {
+        // given
+        String path = "/my/path";
+        String versionName = "latest";
+
+        Resource resource = mockResource(path, versionName, new Date());
+
+        // when
+        PageCompareData pageCompareData = new PageCompareDataImpl(resource, versionName);
+
+        // then
+        assertThat(pageCompareData.getVersions(), not(Collections.<VersionSelection>emptyList()));
+    }
+
+    @Test
+    public void getResource_shouldReturnValue() throws Exception {
+        // given
+        Resource resource = mockResource("/my/path", "latest", new Date());
+
+        // when
+        PageCompareData pageCompareData = new PageCompareDataImpl(resource, "latest");
+
+        // then
+        assertThat(pageCompareData.getResource(), is(resource));
+    }
+
+    @Test
+    public void getVersion() throws Exception {
+        // given
+        String versionName = "latest";
+
+        Resource resource = mockResource("/my/path", versionName, new Date());
+
+        // when
+        PageCompareData pageCompareData = new PageCompareDataImpl(resource, versionName);
+
+        // then
+        assertThat(pageCompareData.getVersion(), is(versionName));
+    }
+
+    @Test
+    public void getVersionDate() throws Exception {
+        // given
+        final Date date = new Date();
+        Resource resource = mockResource("/my/path", "latest", date);
+
+        // when
+        PageCompareData pageCompareData = new PageCompareDataImpl(resource, "latest");
+
+        // then
+        assertThat(pageCompareData.getVersionDate(), is(date));
+    }
+
+    @Test
+    public void getPath() throws Exception {
+        // given
+        final String path = "/my/path";
+        Resource resource = mockResource(path, "latest", new Date());
+
+        // when
+        PageCompareData pageCompareData = new PageCompareDataImpl(resource, "latest");
+
+        // then
+        assertThat(pageCompareData.getPath(), is(path));
+    }
+
+    @Test
+    public void getVersions() throws Exception {
+        // given
+        Resource resource = mockResource("/my/path", "latest", new Date());
+
+        // when
+        PageCompareData pageCompareData = new PageCompareDataImpl(resource, "latest");
+
+        // then
+        assertThat(pageCompareData.getVersions(), not(Collections.<VersionSelection>emptyList()));
+        assertThat(pageCompareData.getVersions().get(1).getName(), is("latest"));
+    }
+
+    @Test
+    public void getLines() throws Exception {
+        // given
+        Resource resource = mockResource("/my/path", "latest", new Date());
```

Comment:
Dont change, just wanted to make you aware if you arent already..

i've recently started using tje new Sling Mocks (and WCM.io AEM mocks: http://wcm.io/testing/aem-mock/) and test writing is a much happier experience!